packages/appstore/src/components/trade-button/trade-button.tsx

```diff
@@ -3,64 +3,66 @@ import { localize } from '@deriv/translations';
 import { Actions } from 'Components/containers/trading-app-card-actions';
 import React from 'react';
 import { Link } from 'react-router-dom';
-import { useStores } from 'Stores';
+import { observer, useStore } from '@deriv/stores';
 
-const TradeButton = ({
-    link_to,
-    onAction,
-    is_external,
-    is_buttons_disabled,
-    new_tab,
-}: Pick<Actions, 'link_to' | 'onAction' | 'is_external' | 'new_tab' | 'is_buttons_disabled'>) => {
-    const { traders_hub, modules } = useStores();
-    const { is_demo } = traders_hub;
-    const { dxtrade_tokens } = modules.cfd;
-    const REAL_DXTRADE_URL = 'https://dx.deriv.com';
-    const DEMO_DXTRADE_URL = 'https://dx-demo.deriv.com';
+const TradeButton = observer(
+    ({
+        link_to,
+        onAction,
+        is_external,
+        is_buttons_disabled,
+        new_tab,
+    }: Pick<Actions, 'link_to' | 'onAction' | 'is_external' | 'new_tab' | 'is_buttons_disabled'>) => {
+        const { traders_hub, modules } = useStore();
+        const { is_demo } = traders_hub;
+        const { dxtrade_tokens } = modules.cfd;
+        const REAL_DXTRADE_URL = 'https://dx.deriv.com';
+        const DEMO_DXTRADE_URL = 'https://dx-demo.deriv.com';
 
-    if (link_to) {
-        if (is_external) {
-            if (new_tab) {
+        if (link_to) {
+            if (is_external) {
+                if (new_tab) {
+                    return (
```

Comment:
can we reduce the cognitive complexity of this code ?

Reply:
I believe that should be address on another card 😃

Reply:
i think its better to change it to a switch case

Reply:
@thisyahlen-deriv I don't think we can make it into a switch case because it is a boolean check so it wouldn't work